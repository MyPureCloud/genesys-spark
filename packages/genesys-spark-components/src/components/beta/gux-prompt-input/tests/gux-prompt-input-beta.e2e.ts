import { newSparkE2EPage, a11yCheck } from '../../../../test/e2eTestUtils';
import { renderConfigs } from './gux-prompt-input-beta.e2e.common';

describe('gux-prompt-input-beta', () => {
  const axeExclusions = [
    {
      issueId: 'color-contrast',
      exclusionReason:
        'COMUI-4091: Ensure the contrast between foreground and background colors meets WCAG 2 AA minimum contrast ratio thresholds for the CTA button (probably just the disabled button state colors need to be changed)'
    }
  ];

  describe('#render', () => {
    renderConfigs.forEach(({ description, html }) => {
      it(description, async () => {
        const page = await newSparkE2EPage({ html });
        const element = await page.find('gux-prompt-input-beta');

        await a11yCheck(page, axeExclusions);

        expect(element.outerHTML).toMatchSnapshot();
      });
    });
  });
});
