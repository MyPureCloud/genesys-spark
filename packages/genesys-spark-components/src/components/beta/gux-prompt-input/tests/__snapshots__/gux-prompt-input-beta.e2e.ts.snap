// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-prompt-input-beta #render should render component as expected 1`] = `
"<gux-prompt-input-beta placeholder="Ask Genesys AI" hydrated=""><template shadowrootmode="open"><div class="gux-input-container"><input class="gux-input" placeholder="Ask Genesys AI" data-testid="prompt-input"><gux-button-slot accent="primary" hydrated=""><button type="button" class="gux-generate" disabled="" title="Generate Genesys AI text" data-testid="generate-button"><gux-icon icon-name="fa/arrow-up-regular" size="small" hydrated=""></gux-icon></button></gux-button-slot></div><div class="caution-message"><slot name="caution-message"></slot></div></template> <div slot="caution-message"> Genesys AI may provide incorrect information. Double-check AI responses.
          Genesys AI privacy, terms and conditions </div> </gux-prompt-input-beta>"
`;
